patcher.sh

```diff
@@ -0,0 +1,330 @@
+set -eou pipefail
+
+# Patching user variables
+OSCHOICE=""
+AM2RZIP=""
+HQMUSIC=false
+SYSTEMWIDE=false
+APPIMAGE=false
+PREFIX=""
+
+# Patching internal variables
+# Since people are likely to double click on this, I need a way to get the script_dir
+dirResult="$(dirname '${BASH_SOURCE[0]}')"
+SCRIPT_DIR="$(realpath $dirResult)"
+VERSION="1.5.5"
+OUTPUT="${SCRIPT_DIR}/am2r_${VERSION}"
+RESOURCES="${OUTPUT}/assets"
+
+patch_am2r ()
+{
+	# Set prefix to default value if empty
+	if [ -z "$PREFIX" ]; then
+		if [ "$SYSTEMWIDE" = true ]; then
+			PREFIX="/usr/local"
+		else
+			PREFIX="$SCRIPT_DIR"
+		fi
+	fi
+
+	# We need to do variable adjustments based on the prefix
+	if [ "$SYSTEMWIDE" = true ]; then
+		OUTPUT="${PREFIX}/opt/am2r"
+		RESOURCES="${OUTPUT}/assets"
+	else
+		OUTPUT="${PREFIX}/am2r_${VERSION}"
+		RESOURCES="${OUTPUT}/assets"
+	fi
+
+	# Cleanup in case the dirs/files exists
+	rm -rf "$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk"
+
+	# Create necessary directories
+	mkdir -p "$OUTPUT" "$RESOURCES"
+
+	# Check for AM2R_11
+	if [[ -f "$AM2RZIP" ]]; then
+		echo "AM2R_11.zip found!"
+		unzip -q "$AM2RZIP" -d "$OUTPUT"
+	elif [[ -d "$AM2RZIP" ]]; then
+		echo "AM2R_11 folder found!"
+		cp -R "$AM2RZIP" "$OUTPUT"
+	else
+		echo "AM2R_11 not found! Please place AM2R_11.zip (case sensitive) into \"$SCRIPT_DIR\" or provide it via command line arguments and try again."
+		return 1
+	fi
+
+	# Check for which OS we patch
+	if [ "$OSCHOICE" = "linux" ]; then
+
+		# Check whether Xdelta is installed
+		if [ ! command -v xdelta3 &> /dev/null ] ; then
+			>&2 echo "Xdelta is not installed! Please install 'xdelta3' from your local package manager!"
+			return 1
+		fi
+
+		echo "Patching for Linux..."
+		echo "Applying AM2R xdelta patch..."
+		xdelta3 -dfs "$OUTPUT/AM2R.exe" "$SCRIPT_DIR/data/AM2R.xdelta" "$OUTPUT/runner"
+
+		echo "Applying data xdelta patch..."
+		xdelta3 -dfs "$OUTPUT/data.win" "$SCRIPT_DIR/data/game.xdelta" "$RESOURCES/game.unx"
+
+		echo "Cleaning up residual AM2R 1.1 files..."
+		rm "$OUTPUT/AM2R.exe" "$OUTPUT/data.win" "$OUTPUT/D3DX9_43.dll"
+
+		echo "Formatting game directory..."
+		# This won't move the runner, because it doesn't have a dot
+		mv $OUTPUT/*.* "$RESOURCES/"
+
+		echo "Installing new datafiles..."
+		cp -R "$SCRIPT_DIR"/data/files_to_copy/* "$RESOURCES/"
+
+		if [ "$HQMUSIC" = true ]; then
+			echo "Copying HQ music..."
+			cp "$SCRIPT_DIR"/data/HDR_HQ_in-game_music/*.ogg "$RESOURCES/"
+		fi
+
+		# Format music for Unix, aka lowercase it
+		# Finds every file, and if its not already lowercase, lowercases it.
+		find "$RESOURCES" -type f -exec bash -c '
+		    target="{}";
+		    cd "$(dirname "${target}")";
+		    target="$(basename "${target}")";
+		    ! [[ "${target}" = "${target,,}" ]] && mv "${target}" "${target,,}"
```

Comment:
This is because of my previous CR comment being wrong, but the semicolons arent needed:
```suggestion
                # On Unix the music filenames need to be converted to lowercase.
		find "$RESOURCES" -type f -exec bash -c '
		    target="{}"
		    # Only files are meant to be modified, not the folders they're contained in.
		    cd "$(dirname "${target}")"
		    target="$(basename "${target}")"
		    # Convert the filename to lowercase, if required.
		    ! [[ "${target}" = "${target,,}" ]] && mv "${target}" "${target,,}"
```
Also breaks up the comment a bit

Reply:
Brought in with some slight grammar change so that Kate doesn't complain.